presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -7549,6 +7549,80 @@ public void testSelectFromViewWithoutDefaultCatalogAndSchema()
+        assertUpdate(defaultSession, "CREATE VIEW " + prestoViewNameDefault + " AS SELECT *  FROM " + hiveTableName);
+
+        assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123')");
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () ->assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.tpch." + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123456789')")
+        ).hasMessageContaining("Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):\n    [1990-01-02 12:13:14.123]");
+
+        assertQuery(millisSession, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123')");
+        assertQuery(millisSession, "SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.tpch." + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123')");
+
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () -> assertQuery(nanosSessions, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123456789')")
+        ).hasMessageContaining("Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):\n    [1990-01-02 12:13:14.123]");
+
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () -> assertQuery(nanosSessions, "SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.tpch." + prestoViewNameDefault, "VALUES ('1990-01-02 12:13:14.123456789')")
+        ).hasMessageContaining("Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):\n    [1990-01-02 12:13:14.123]");
+
+        // Presto view created with config property set to MILLIS and session property set to NANOS
+        String prestoViewNameNanos = "select_from_presto_view_hive_table_timestamps_millis_view_" + randomTableSuffix();
+        assertUpdate(nanosSessions, "CREATE VIEW " + prestoViewNameNanos + " AS SELECT *  FROM " + hiveTableName);
+
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () -> assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM " + prestoViewNameNanos, "VALUES ('1990-01-02 12:13:14.123')")
+        ).hasMessageContaining("Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):\n    [1990-01-02 12:13:14.123000000]");
+        assertThatThrownBy(
+                // TODO Presto view schema is fixed on creation
+                () ->assertQuery(defaultSession, "SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.tpch." + prestoViewNameNanos, "VALUES ('1990-01-02 12:13:14.123456789')")
+        ).hasMessageContaining("Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):\n    [1990-01-02 12:13:14.123000000]");
```

Comment:
Quote both expected and actual within `hasMessageContaining`

or, use the actual wrong value in assertion and put the correct value in a TODO comment

Reply:
Same question as above

Reply:
Migrated to "correct value in TODO". After migration to assertThat and getting rid fo varchar the assertion started complaining about type mismatch, not value mismatch, which was not informative enough.